apps/server/src/shared/domain/entity/base.entity.ts

```diff
@@ -0,0 +1,27 @@
+import { PrimaryKey, Property, SerializedPrimaryKey } from '@mikro-orm/core';
+import { ObjectId } from '@mikro-orm/mongodb';
+
+export abstract class BaseEntity {
+	@PrimaryKey()
+	_id!: ObjectId;
+
+	@SerializedPrimaryKey()
+	id!: string;
+}
+
+// NOTE apparently we can only have one step of inheritance
+// that's why we definethe whole class here
+// TODO implement cutom class decorators
+export abstract class BaseEntityWithTimestamps {
```

Comment:
Or both classes could implement I base entity?

Reply:
we actually had an issue with inheritance when writing tests. maybe we should look into custom class decorators.